pkg/extensions/search/common/oci_layout.go

```diff
@@ -19,21 +20,22 @@ import (
 
 // OciLayoutInfo ...
 type OciLayoutUtils struct {
-	Log log.Logger
+	Log             log.Logger
+	StoreController storage.StoreController
 }
 
 // NewOciLayoutUtils initializes a new OciLayoutUtils object.
-func NewOciLayoutUtils(log log.Logger) *OciLayoutUtils {
-	return &OciLayoutUtils{Log: log}
+func NewOciLayoutUtils(storeController storage.StoreController, log log.Logger) *OciLayoutUtils {
+	return &OciLayoutUtils{Log: log, StoreController: storeController}
```

Comment:
Do we need StoreController here ? 

I was thinking to keep oci layout utils independent of store controller and whoever wants to use  utils function they just need to provide correct path.

Reply:
Yes, I saw your refactor, unfortunately I need to use imageStore.GetIndexContent() and imageStore.GetBlobContent() inside oci_layout.
If you think about another solution, let me know.

The only thing I can think about is passing storeController from resolver.go down to each function.

Reply:
This is the approach I am using in #233 , passing storeController to each function.